RailgunsRetargeting/RailgunsRetargeting.cs

```diff
@@ -65,18 +66,21 @@ internal static int GetEjectorUID(EjectorComponent ejector)
 
         internal static ManagedEjector GetOrCreateManagedEjector(int ejectorUID, int originalOrbitId)
         {
+            managedEjectors.TryGetValue(ejectorUID, out ManagedEjector value);
 
-            if (!managedEjectors.ContainsKey(ejectorUID))
+            if (value == null)
```

Comment:
This would ensure the data would be consistent.

```csharp
if (!managedEjectors.TryGetValue(ejectorUID, out ManagedEjector value))
{
   ...
}

Reply:
There is a slight problme here, GetOrCreateManagedEjector would possibly return outdated data here. There is no synchronization for accessing the value at callee site.